File: 14-longest-common-prefix/14-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        ans = []
        n = len(strs)
        
        strs.sort()
        
        for i in range(len(strs[0])):
            if strs[0][i] == strs[n - 1][i]:
                ans.append(strs[0][i])
            else:
                break
            
        return "".join(ans)

File: 14-longest-common-prefix/14-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        return solve(strs);
    }
private:
    // O(nlogn) Time and O(1) extra space
    string solve(vector<string>& strs) {
        int n = strs.size();
        
        sort(strs.begin(), strs.end());
        
        string ans = "";
        for(int i = 0; i < strs[0].size(); ++i) {
            if(strs[0][i] == strs[n - 1][i])
                ans += strs[0][i];
            else
                break;
        }
        return ans;
    }
};